diffrax/_brownian/tree.py

```diff
@@ -63,62 +64,113 @@
 _BrownianReturn = TypeVar("_BrownianReturn", bound=AbstractBrownianIncrement)
 
 
+# An internal dataclass that holds the rescaled Levy areas
+# in addition to the non-rescaled ones, for the purposes of
+# taking the difference between two Levy areas.
+class _AbstractLevyVal(eqx.Module):
+    dt: eqx.AbstractVar[Inexact[Array, ""]]
+    W: eqx.AbstractVar[Array]
+
+
+class _BMLevyVal(_AbstractLevyVal):
+    dt: Inexact[Array, ""]
+    W: Array
+
+
+class _AbstractSpaceTimeLevyVal(_AbstractLevyVal):
+    H: eqx.AbstractVar[Array]
+    bar_H: eqx.AbstractVar[Array]
+
+
+class _SpaceTimeLevyVal(_AbstractSpaceTimeLevyVal):
+    dt: Inexact[Array, ""]
+    W: Array
+    H: Array
+    bar_H: Array
+
+
+class _SpaceTimeTimeLevyVal(_AbstractSpaceTimeLevyVal):
+    dt: Inexact[Array, ""]
+    W: Array
+    H: Array
+    bar_H: Array
+    K: Array
+    bar_K: Array
```

Comment:
Possibly slightly simpler might be something like:
```python
class _LevyVal(eqx.Module):
    brownian_increment: AbstractBrownianIncrement
    bar_H: Optional[Array]
    bar_K: Optional[Array]
```
?
Possibly adding some `Generic[...]` typevars for each of the three arguments if required.

Reply:
Fair enough, I suppose I was trying to do it very properly, but since it's never exposed to the user, you're right, I should go for the simpler option.